falcon/response.py

```diff
@@ -136,12 +144,118 @@ def set_stream(self, stream, stream_len):
         self.stream = stream
         self.stream_len = stream_len
 
+    def set_cookie(self, name, value, expires=None, max_age=None,
+                   domain=None, path=None, secure=True, httponly=True):
+        """Set a response cookie.
+
+        Note:
+            :meth:`~.set_cookie` is capable of setting multiple cookies
+            with specified cookie properties on the same request.
+            Currently :meth:`~.set_header` and :meth:`~.append_header` are
+            not capable of this.
+
+        Args:
+            name (str):
+                Cookie name
+            value (str):
+                Cookie value
+            expires (datetime): Defines when the cookie should expire.
+                the default is to expire when the browser is closed.
+            max_age (int): The Max-Age attribute defines the lifetime of the
+                cookie, in seconds.  The delta-seconds value is a decimal non-
+                negative integer.  After delta-seconds seconds elapse,
+                the client should discard the cookie.
+            domain (str): The Domain attribute specifies the domain for
+                which the cookie is valid.
+                An explicitly specified domain must always start with a dot.
+                A value of 0 means the cookie should be discarded immediately.
+            path (str): The Path attribute specifies the subset of URLs to
+                which this cookie applies.
+            secure (bool) (default: True): The Secure attribute directs the
+                user agent to use only secure means to contact the origin
+                server whenever it sends back this cookie.
+                Warning: You will also need to enforce HTTPS for the cookies
+                to be transfered securely.
+            httponly (bool) (default: True):
+                The attribute httponly specifies that the cookie
+                is  only transferred in HTTP requests, and is not accessible
+                through JavaScript. This is intended to mitigate some forms
+                of cross-site scripting.
+        Note:
+            Kwargs and their valid values are all
+            specified in http://tools.ietf.org/html/rfc6265
+
+        See Also:
+            :ref:`Setting Cookies <setting-cookies>`
+
+        Raises:
+            KeyError if ``name`` is not a valid cookie name.
+
+        """
+
+        if six.PY2:  # pragma: no cover
+            # NOTE(tbug): In python 2 we want a str type name,
+            # as unicode will make the SimpleCookie implementation
+            # blow up.
+            if isinstance(name, unicode):
+                try:
+                    name = name.encode("ascii")
+                except UnicodeEncodeError as e:
+                    raise KeyError("'name' must consist of only ascii")
+
+        if self._cookies is None:
+            self._cookies = SimpleCookie()
+
+        try:
+            self._cookies[name] = value
```

Comment:
If the value is unicode outside of the ascii range, it'll raise a UnicodeEncodeError just like the name does. As a result, it would probably be good to handle that and add a couple tests to cover that scenario.